va/validation-authority.go

```diff
@@ -275,6 +279,14 @@ func (va *ValidationAuthorityImpl) fetchHTTP(ctx context.Context, identifier cor
 			Detail: fmt.Sprintf("Error reading HTTP response body: %v", err),
 		}
 	}
+	// io.LimitedReader will silently truncate a Reader so if the
+	// resulting payload is the same size as maxResponseSize fail
+	if len(body) >= maxResponseSize {
+		return nil, validationRecords, &probs.ProblemDetails{
+			Type:   probs.UnauthorizedProblem,
+			Detail: fmt.Sprintf("Invalid response from %s: response too large, \"%s\"", url.String(), body),
```

Comment:
I would leave out "response too large" here. In 99.99% of cases the problem will be that the response just doesn't match. Adding "response too large" is confusing and distracting.